### PR TITLE
Add BatchNorm information collection

### PR DESCRIPTION
**Pull request by lapid92**

### CODE REVIEW

File: model_compression_toolkit/common/substitutions/batchnorm_info_collection.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        """
+        Initialize a BatchNormalizationInformation object.
+
+        Args:
+            gamma: BN scaling factor.
+            beta: BN offset factor.
+            moving mean: Moving mean of the batches.
+            moving variance: Moving variance of the batches.
+            epsilon: constant added to batches variance.
+        """
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/substitutions/batchnorm_info_collection.py:45]: Fix args names (moving mean -> moving_mean, etc.)

File: model_compression_toolkit/common/graph/graph_matchers.py [hunk 2]
```diff
@@ -52,6 +51,24 @@ def apply(self, input_node_object: Any) -> bool:
             return True
 
 
+class EveryNodeOperationMatcher(node_matcher.BaseNodeMatcher):
```
Comment (haihabi) [inline model_compression_toolkit/common/graph/graph_matchers.py:54]: Remove this class and use Node Matcher instade.

File: model_compression_toolkit/common/post_training_quantization.py [hunk 3]
```diff
@@ -371,25 +366,27 @@ def _prepare_model_for_quantization(in_model: Any,
...
-        tb_w.add_graph(transformed_graph, 'pre_statistics_collection_substitutions')
+    graph = substitute(graph, fw_impl.get_substitutions_prepare_graph())
 
     #########################################
     # Set prior info to nodes
     ##########################################
-    for node in transformed_graph.nodes:
+    for node in graph.nodes:
         node.prior_info = fw_impl.get_node_prior_info(node=node,
                                                       fw_info=fw_info)
```
Comment (haihabi) [inline model_compression_toolkit/common/post_training_quantization.py:382]: Why does the batch-normalization information is not initialized in get_node_prior_info?

File: model_compression_toolkit/common/post_training_quantization.py [hunk 4]
```diff
@@ -368,25 +366,27 @@ def _prepare_model_for_quantization(in_model: Any,
...
 
     ######################################
-    # Graph substitution (pre statistics collection)
+    # Graph substitution (prepare graph)
     ######################################
-    transformed_graph = substitute(graph, fw_impl.get_substitutions_pre_statistics_collection())
-
-    if tb_w is not None:
-        tb_w.add_graph(transformed_graph, 'pre_statistics_collection_substitutions')
+    graph = substitute(graph, fw_impl.get_substitutions_prepare_graph())
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/post_training_quantization.py:375]: I wouldn't remove the logger after the prepare part...

File: tests/keras_tests/function_tests/test_bn_info_collection.py [hunk 5]
```diff
@@ -0,0 +1,457 @@
+# Copyright 2021 Sony Semiconductors Israel, Inc. All rights reserved.
+#
```
Comment (reuvenperetz) [inline tests/keras_tests/function_tests/test_bn_info_collection.py:2]: Please use license with 2022 for new files

File: model_compression_toolkit/keras/keras_node_prior_info.py [hunk 6]
```diff
@@ -58,3 +68,30 @@ def _get_min_max_outputs(node: BaseNode,
...
+
+    Returns:
+        Mean/Std output values if known.
+    """
+    mean_output, std_output = None, None
+
+    if node.type == BatchNormalization:
+        std_output = node.get_weights_by_keys(GAMMA)
+        mean_output = node.get_weights_by_keys(BETA)
+
```
Comment (reuvenperetz) [inline model_compression_toolkit/keras/keras_node_prior_info.py:88]: If we don't have them use 0,1

**Review by reuvenperetz — CHANGES_REQUESTED** [review]
